File: python/enums1.py
class Direction:
    pass


class Up(Direction):
    pass


class Down(Direction):
    pass


class Left(Direction):
    pass


class Right(Direction):
    pass


type_to_arrow = {
    Up: '↑',
    Down: '↓',
    Left: '←',
    Right: '→',
}


def to_arrow(dir: Direction) -> str:
    return type_to_arrow[type(dir)]


if __name__ == '__main__':
    print(to_arrow(Up()))
    print(to_arrow(Down()))
    print(to_arrow(Left()))
    print(to_arrow(Right()))

File: rust/enums1.rs
enum Direction {
    Up,
    Down,
    Left,
    Right,
}

fn to_arrow(dir: &Direction) -> String {
    let s = match dir {
        Direction::Up => "↑",
        Direction::Down => "↓",
        Direction::Left => "←",
        Direction::Right => "→",
    };

    s.into()
}


fn main() {
    println!("{}", to_arrow(&Direction::Up));
    println!("{}", to_arrow(&Direction::Down));
    println!("{}", to_arrow(&Direction::Left));
    println!("{}", to_arrow(&Direction::Right));
}
